server/src/main/java/io/crate/planner/node/ddl/AlterTableAddColumnPlan.java

```diff
@@ -100,13 +102,17 @@ public void executeOrFail(DependencyCarrier dependencies,
                 tableElements
             );
 
-            List<Reference> references = new ArrayList<>();
+            // We can add multiple object columns via ALTER TABLE ADD COLUMN.
+            // Those columns can have overlapping paths, for example we can add columns o['a']['b'] and o['a']['c'].
+            // For every added column AnalyzedColumnDefinition provides not only leaf but also path to the root.
+            // For o['a']['b'] and o['a']['c'] we can end up having Ref(Ident(o)) and Ref(Ident(a)) twice.
+            Set<Reference> references = new HashSet<>();
             IntArrayList pKeysIndices = new IntArrayList();
             tableElements.collectReferences(alterTable.tableInfo().ident(), references, pKeysIndices, true);
 
             var addColumnRequest = new AddColumnRequest(
                 alterTable.tableInfo().ident(),
-                references,
+                new ArrayList<>(references), // We don't use Set in the request itself since we need directly indexed structure referred by pKeysIndices.
```

Comment:
Can we have immutable list here as well?

Reply:
Could be but also might be unnesesary overhead for now as current usages of `references()` are read only and supposed to be changed rarely. No strong opinion here, updated in https://github.com/crate/crate/pull/13383/commits/34c3244c0cb2a4a3f956678546f570bb9a34c7e5